server/internal/openapi/process.go

```diff
@@ -138,6 +138,30 @@ func (p *ToolExtractor) Do(
 		return nil, oops.E(oops.CodeInvariantViolation, oops.Permanent(err), "unable to process openapi document").Log(ctx, p.logger)
 	}
 
+	var (
+		doc     []byte
+		readErr error
+	)
+	func() {
+		rc, err := p.assetStorage.Read(ctx, docURL)
+		if err != nil {
+			readErr = oops.E(oops.CodeUnexpected, err, "error fetching openapi document").Log(ctx, p.logger)
+			return
+		}
+		defer o11y.LogDefer(ctx, p.logger, func() error {
+			return rc.Close()
+		})
+
+		doc, err = io.ReadAll(rc)
+		if err != nil {
+			readErr = oops.E(oops.CodeUnexpected, err, "error reading openapi document").Log(ctx, p.logger)
+			return
+		}
+	}()
+	if readErr != nil {
+		return nil, fmt.Errorf("failed to read openapi document: %w", readErr)
```

Comment:
We shouldn't mix regular errors with oops.E's. Usually if you see oops in a function then stick with that.
Truth be told, this package doesn't need to be using oops but we should refactor all or none.